backend/core/src/auth/passport-strategies/local-mfa.strategy.ts

```diff
@@ -31,31 +30,31 @@ export class LocalMfaStrategy extends PassportStrategy(Strategy, "localMfa") {
     const validationPipe = new ValidationPipe(defaultValidationPipeOptions)
     const loginDto: LoginDto = await validationPipe.transform(req.body, {
       type: "body",
-      metatype: LoginDto,
+      metatype: LoginDto
     })
 
     const user = await this.userRepository.findOne({
       where: { email: loginDto.email.toLowerCase() },
-      relations: ["leasingAgentInListings"],
+      relations: ["leasingAgentInListings"]
     })
 
     if (user) {
-      if (user.lastLoginAt) {
+      if (user.hasLoggedInAtLeastOnce() && user.isPartnerOrAdmin()) {
```

Comment:
Why does it matter if the user has logged in at least once before? Would it be more appropriate to check if the user is confirmed?